### PR TITLE
Try out TypeGuard for ScopeTreeNodeWithPathId

### PR DESCRIPTION
**Pull request by msullivan**

### CODE REVIEW

**Comment by msullivan** [comment]
First take of this failed because the typing_extensions that the edgedb python package depends on is fixed to 3.7.4; I sent a PR there to bump that (and then do we need to do a version bump?), but we can also just if TYPE_CHECKING the import here, so.

File: edb/ir/scopetree.py [hunk 1]
```diff
@@ -21,6 +21,8 @@
 
 from __future__ import annotations
 from typing import *
+if TYPE_CHECKING:
+    from typing_extensions import TypeGuard
```
Comment (elprans) [inline edb/ir/scopetree.py:25]: Don't forget to add a dependency to `setup.py`

Comment (msullivan) [reply edb/ir/scopetree.py:25]: The stubs are mypy internal, and we aren't actually importing it yet at runtime

**Comment by msullivan** [comment]
Any reason not to merge this, since it doesn't actually need the runtime dep?

**Review by elprans — APPROVED** [review]

**Pull request merged**